app/lib/i18n/utils.ts

```diff
@@ -0,0 +1,21 @@
+import {
+    DEFAULT_LOCALE,
+    SUPPORTED_LOCALES
+} from '~/constants';
+
+export const extractLocaleFromPath = (
+    pathname: string
+): string | null => {
+    const segments = pathname.split('/');
```

Comment:
You don't need to split all of the URL into parts; you need only the second part. So `split` has the second param for that - stop after a second part.
Since middleware should be as fast as possible, I think it's a nice little improvement in performance.
```suggestion
    const segments = pathname.split('/', 2);
```

Reply:
I agree with you, thanks